File: code2map/docs/examples/python/output/parts/UserManagementService_find_by_age_range.py
# code2map fragment (non-buildable)
# original: docs/examples/python/user_management_service.py
# lines: 160-175
# symbol: UserManagementService#find_by_age_range
# notes: references dataclasses.dataclass, re, typing.Optional; calls result.append, self._users.values
    def find_by_age_range(self, min_age: int, max_age: int) -> list[User]:
        """
        年齢範囲でユーザーを検索する

        Args:
            min_age: 最小年齢（含む）
            max_age: 最大年齢（含む）

        Returns:
            条件に一致するユーザーのリスト
        """
        result = []
        for user in self._users.values():
            if min_age <= user.age <= max_age:
                result.append(user)
        return result

File: docs/examples/java/output/parts/UserManagementService_findByAgeRange.java
// code2map fragment (non-buildable)
// original: docs/examples/java/UserManagementService.java
// lines: 130-138
// symbol: UserManagementService#findByAgeRange
// notes: references java.util.ArrayList, java.util.HashMap, java.util.List, java.util.Map, java.util.Optional, java.util.regex.Pattern; calls result.add, user.getAge, users.values
    public List<User> findByAgeRange(int minAge, int maxAge) {
        List<User> result = new ArrayList<>();
        for (User user : users.values()) {
            if (user.getAge() >= minAge && user.getAge() <= maxAge) {
                result.add(user);
            }
        }
        return result;
    }
